src/history/hs_cursor.c

```diff
@@ -48,9 +48,19 @@ __wt_hs_row_search(WT_CURSOR_BTREE *hs_cbt, WT_ITEM *srch_key, bool insert)
         WT_WITH_BTREE(CUR2S(hs_cbt), CUR2BT(hs_cbt),
           ret = __wt_row_search(hs_cbt, srch_key, insert, NULL, false, NULL));
 
+    if (ret == 0 && !insert) {
+        WT_ERR(__wt_key_return(hs_cbt));
+        WT_ERR(__wt_value_return(hs_cbt, hs_cbt->upd_value));
+    }
+
 #ifdef HAVE_DIAGNOSTIC
-    WT_TRET(__wt_cursor_key_order_init(hs_cbt));
+    WT_ERR(__wt_cursor_key_order_init(hs_cbt));
```

Comment:
why we need to overwrite the error in diagnostic mode

Reply:
I think we should fail at this spot if this check fail.

Reply:
Also this is now not the last statement in this function.